src/rules/no-missing-label-refs.js

```diff
@@ -40,7 +40,8 @@ function findMissingReferences(node, nodeText) {
 	 * `left` is the content between the first brackets. It can be empty.
 	 * `right` is the content between the second brackets. It can be empty, and it can be undefined.
 	 */
-	const labelPattern = /\[(?<left>[^\]]*)\](?:\[(?<right>[^\]]*)\])?/dgu;
+	const labelPattern =
+		/(?<!\\)\[(?<left>[^\]]*)(?<!\\)\](?<!\\)(?:\[(?<right>[^\]]*)(?<!\\)\])?/dgu;
```

Comment:
This is still not working well when the label itself contains escaped brackets, for example `"[\\[foo\\]]"` should trigger a violation, but it doesn't. I think we could fix that by improving the subpatterns where labels are matched. Maybe like this, or similar:

```suggestion
	const labelPattern =
		/(?<!\\)\[(?<left>(?:\\.|[^\]])*)(?<!\\)\](?<!\\)(?:\[(?<right>(?:\\.|[^\]])*)(?<!\\)\])?/dgu;
```

Reply:
@fasttime 

Thanks! I shoud have looked more closely at the edge cases.

I've added more test cases and updated the regex in https://github.com/eslint/markdown/pull/348/commits/2f2dc47a7f89e8c55e83194399c10e262c8f88cc.